documentation/modules/ROOT/pages/reactive-messaging.adoc

```diff
@@ -0,0 +1,365 @@
+                .onOverflow().drop() //<2>
+                .map(tick -> beers.get(ThreadLocalRandom.current().nextInt(0, beers.size()))) //<3>
+                .map(JsonbBuilder.create()::toJson); //<4>
+    }
+
+    @Incoming("beers")
+    @Outgoing("groups")
+    public Multi<List<String>> skipGroup(Multi<String> stream) {
+        return stream.skip().first(Duration.ofMillis(10)).group().intoLists().of(5); //<5>
+    }
+
+
+    @Incoming("groups")
+    @Outgoing("messages")
+    public String processGroup(List<String> list) {
+        return String.join(",", list.toString()); //<6>
+    }
+
+
+    @Incoming("messages")
+    public String print(String msg) {
+        System.out.println(msg);
+        return  msg; //<7>
+    }
+}
+----
+<1> We're creating a Multi that generates a new message every `1` second.
+<2> We apply backpressure by dropping the messages if the topic is not ready.
+<3> For each message we choose a random `Beer` from our list.
+<4> We map the `Beer` to JSON format.
+<5> The method `skipGroup' skips the messages sent during the first 10 milliseconds and groups next into lists of 5.
+<6> The `processGroup` method takes each group and does further processing.
+<7> The `print` method is just to have a manner to observe the output.
+
+== Block Processing
+
+Invoking a blocking operation/service can take longer time. In such cases you can protect your implementation
+by using `@Blocking` annotation. 
+Let's modify the `BeerGenerator` Java class in `src/main/java` in the `com.redhat.developers` package 
+by adding the `@Blocking` annotation to the `processGroup` method:
+
+[.console-input]
+[source,java]
+----
+package com.redhat.developers;
+
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.json.bind.JsonbBuilder;
+
+import io.smallrye.common.annotation.Blocking;
+import io.smallrye.mutiny.Multi;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+
+@ApplicationScoped
+public class BeerGenerator {
+
+    @RestClient
+    BeerService service;
+
+    @Outgoing("beers")
+    Multi<String> beers() {
+        List<Beer>  beers = service.getBeers(10);
+        return Multi.createFrom().ticks().every(Duration.ofSeconds(1))
+                .onOverflow().drop()
+                .map(tick -> beers.get(ThreadLocalRandom.current().nextInt(0, beers.size())))
+                .map(JsonbBuilder.create()::toJson);
+    }
+
+    @Incoming("beers")
+    @Outgoing("groups")
+    public Multi<List<String>> group(Multi<String> stream) {
+        return stream.skip().first(Duration.ofMillis(10)).group().intoLists().of(5);
+    }
+
+
+    @Incoming("groups")
+    @Outgoing("messages")
+    @Blocking
+    public String processGroup(List<String> list) {
```

Comment:
What the idea of this method ? Just to show that Mutiny can handle blocking logic as well ? Shouldn't we rename the method to something like `processGroupBlocking` ?

Reply:
It is more to showcase that when you have a blocking operation within your method's logic, you can signal that with Blocking annotation. In this case I am not using Mutiny.

Reply:
make sense